src/MonadLoggerEffect.hs

```diff
@@ -0,0 +1,33 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE GHC2021 #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-} -- Needed for 'MonadLogger (Eff es)'
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module MonadLoggerEffect (MonadLoggerEffect (..), runLoggerStdout) where
+
+import Control.Monad.Logger (Loc, LogLevel, LogSource, MonadLogger (..), ToLogStr (toLogStr),
+                             defaultOutput)
+import Effectful (Dispatch (Dynamic), DispatchOf, Eff, Effect, IOE, MonadIO (liftIO), (:>))
+import Effectful.Dispatch.Dynamic (interpret, send)
+import System.IO (stdout)
+
+data MonadLoggerEffect :: Effect where
+    MonadLoggerLog :: ToLogStr msg => Loc -> LogSource -> LogLevel -> msg -> MonadLoggerEffect m ()
```

Comment:
For this I was really happy with the guide in the documentation:

https://hackage.haskell.org/package/effectful-core-2.2.2.2/docs/Effectful-Dispatch-Dynamic.html#g:4

It matched this use case precisely! The nice thing is that I didn't have to edit any of the code that called `monad-logger`'s log functions. The downside is that I need an orphan instance here.

I think it's worth it, though, because this makes `monad-logger` fit really snugly in the `effectful` ecosystem. You can call them if you have the `MonadLoggerEffect` effect. Makes perfect sense!

Reply:
There's no need to replace monad-logger right away and for the sake of keeping the diffs readable adapting `monad-logger` this way is great, but would it make sense to move to another logger that directly uses effectful's effect system at some point? A cursory search brought up [log-effectful](https://hackage.haskell.org/package/log-effectful).

Reply:
We could move, though if you look at the source code of that library, you'll see that it applies the exact same pattern as I do here, namely creating an orphan instance of the library's monad class.